pkg/kubelet/network/dns/dns.go

```diff
@@ -99,6 +100,14 @@ func omitDuplicates(strs []string) []string {
 func (c *Configurer) formDNSSearchFitsLimits(composedSearch []string, pod *v1.Pod) []string {
 	limitsExceeded := false
 
+	for i, search := range composedSearch {
```

Comment:
does this mean 1.21 and 1.22 kubelets can break when running a pod with expanded DNS config allowed by a 1.23 API server?

Reply:
Not exactly. 
A pod's search path will be validated by the API server to not exceed 253 characters, but kubelet's resolv.conf can contain a search path of more than 253 characters at 1.22 kubelet. 1.22 kubelet has to validate this.